relay-server/src/services/project.rs

```diff
@@ -692,39 +692,41 @@ impl Project {
         envelope_processor: Addr<EnvelopeProcessor>,
         buckets: Vec<Bucket>,
     ) {
-        if self.metrics_allowed() {
-            match &mut self.state {
-                State::Cached(state) => {
-                    let state = Arc::clone(state);
-
-                    let buckets =
-                        self.rate_limit_metrics(&state, buckets, outcome_aggregator.clone());
-
-                    if !buckets.is_empty() {
```

Comment:
I would keep this. Returns in already (deeply) nested paths mean no other later code is run. E.g. if we add code after the match statement (logging, other logic) this wouldn't run anymore. 

So in order to understand if this return is correct I first need to check the rest of the function and evaluate if this is still correct behaviour.